pkg/auth/model/model.go

```diff
@@ -199,6 +199,11 @@ type CredentialKeys struct {
 	SecretAccessKey string `json:"secret_access_key"`
 }
 
+type ExternalPrincipal struct {
+	ID     string `db:"id"`
+	UserID string `db:"user_id"`
+}
+
```

Comment:
Yay! Another 2 `id`s in auth package!!
Document these values carefully, what are the possible values for them. Even before E15l, it's already extermelllllly complicated.

Also - do you need the `db` notations?

Reply:
updated the struct with comment and correct tag notations.